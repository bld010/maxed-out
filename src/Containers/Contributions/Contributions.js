import React, { Component} from 'react';
import { connect } from 'react-redux';
import './Contributions.scss';
import { setPacContributions } from '../../actions/index'

export class Contributions extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: ''
    }
  }

  generateContributionList = () => {
    if (this.props.type === 'PAC' && this.props.pacContributions.length === 0 && this.props.candidate !== null) {
      return <h4>No {this.props.type} contributions found.</h4>
    }
    if (this.props.type === 'Individual' && this.props.individualContributions.length === 0 && this.props.candidate !== null) {
      return <h4>No {this.props.type} contributions found.</h4>
    }

    if( this.props.type === 'PAC' && this.props.pacContributions.length > 0) {
      return this.props.pacContributions.map((contribution, index) => {
        return <div className="contribution" key={index}>
          <h4>{contribution.contributor_name}: ${contribution.contribution_receipt_amount} ({contribution.two_year_transaction_period})</h4>
          <p>Type: {contribution.line_number_label}</p>
          <p>Memo: {contribution.memo_text}</p>
        </div>
      })
    }

    if ( this.props.type === 'Individual' && this.props.pacContributions.length > 0) {
      return this.props.individualContributions.map((contribution, index) => {
        return <div className="contribution" key={index}>
            <h4>{contribution.contributor_name}: $ {contribution.contribution_receipt_amount} ({contribution.two_year_transaction_period})</h4>
            <p>Employer: {contribution.contributor_employer} | Location: {contribution.contributor_city}, {contribution.contributor_state}
            </p>
           
          </div>
      })
    }
  }

  render() {
    return(
      <section className="Contributions">
        {this.generateContributionList(this.props.type)}
      
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  candidate: state.candidate,
  committee_id: state.committee_id,
  pacContributions: state.pacContributions,
  individualContributions: state.individualContributions
})

export default connect(mapStateToProps)(Contributions)


// propTYpes
// committee_id, type, candidate, setPacCOntributions


