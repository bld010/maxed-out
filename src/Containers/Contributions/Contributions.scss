.Contributions {
  width: 100%;
  background: white;
  min-height: 200px;
  overflow: scroll;
  filter: drop-shadow(0 0 8px #333);
  border-radius: 8px;
  margin: 10px;
  max-height: 90%;
  padding: 20px;
  will-change: transform;


  & h3 {
    font-size:18;
    margin: 0 0 10px 5px;
    border-bottom: 1px solid black;
    position: relative;
    z-index: 1;
    background: white;
    color: rgb(98, 84, 129)
  }

  & .contribution {
    margin: 5px;
    font-size: 14px;
    text-align: left;
    padding: 5px;
    border-bottom: 1px solid black;
    text-transform: capitalize;
    
    
    & h4 {
      font-size: 16px;
      margin: 5px 0 0 0;
    }

    & p {
      margin: 4px 0 4px 0;
    }
  }
}