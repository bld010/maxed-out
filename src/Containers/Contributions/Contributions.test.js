import React from 'react';
import { shallow } from 'enzyme';

import { Contributions, mapStateToProps } from './Contributions'

describe('Contributions', () => {

  it('should match the snapshot with correct data passed in', () => {
    let wrapper = shallow(
    <Contributions
      type="PAC"
      pacContributions={[]}
      candidate={{name: 'Liz'}}
    />)

    expect(wrapper).toMatchSnapshot();
  })

  describe('generateContributionList', () => {

    describe('PAC contributions', () => {
      
      let mockContributions = [
        { contributor_name: 'David PAC', 
          contribution_receipt_amount: 234234,
          two_year_transaction_period: 2020,
          line_number_label: 'Memo',
          memo_text: 'This is a donation' },
        { contributor_name: 'Chris PAC', 
        contribution_receipt_amount: 234,
        two_year_transaction_period: 2020,
        line_number_label: 'Memo',
        memo_text: 'This is a donation' },
        { contributor_name: 'David & Chris PAC', 
        contribution_receipt_amount: 2,
        two_year_transaction_period: 2020,
        line_number_label: 'Memo',
        memo_text: 'This is a donation' }
      ]

      it('should return a no results message if no contributions, but candidate is found', () => {
        let wrapper = shallow(
          <Contributions
            type="PAC"
            pacContributions={[]}
            candidate={{name: 'Liz'}}
          />)

        let expected = 'No PAC contributions found for the 2020 election cycle.'
      
        let result = wrapper.instance().generateContributionList().props.children.join('')
        expect(result).toEqual(expected)
        
      })

      it('should return an array of JSX divs if contributions length is greater than 0', () => {
        let wrapper = shallow(
          <Contributions
            type="PAC"
            pacContributions={mockContributions}
            candidate={{name: 'Liz'}}
          />)

          let result = wrapper.instance().generateContributionList();
          expect(result.length).toEqual(3)
          expect(result[0].props.children[0].props.children[0]).toEqual('david pac')
      })
    })

    describe('Individual contributions', () => {
      
      let mockContributions = [
        { contributor_name: 'David PAC', 
          contribution_receipt_amount: 234234,
          two_year_transaction_period: 2020,
          contributor_state: 'CO',
          contributor_city: 'denver',
          contributor_employer: 'self employed'},
          { contributor_name: 'Chris PAC', 
          contribution_receipt_amount: 234,
          two_year_transaction_period: 2020,
          contributor_state: 'CO',
          contributor_city: 'denver',
          contributor_employer: 'self employed'},
          { contributor_name: 'David & Chris PAC', 
          contribution_receipt_amount: 2,
          two_year_transaction_period: 2020,
          contributor_state: 'CO',
          contributor_city: 'denver',
          contributor_employer: 'self employed'}
      ]

      it('should return a no results message if no contributions, but candidate is found', () => {
        let wrapper = shallow(
          <Contributions
            type="Individual"
            individualContributions={[]}
            candidate={{name: 'Liz'}}
          />)

        let expected = 'No Individual contributions found for the 2020 election cycle.'
      
        let result = wrapper.instance().generateContributionList().props.children.join('')
        expect(result).toEqual(expected)
        
      })

      it('should return an array of JSX divs if contributions length is greater than 0', () => {
        let wrapper = shallow(
          <Contributions
            type="Individual"
            individualContributions={mockContributions}
            candidate={{name: 'Liz'}}
          />)

          let result = wrapper.instance().generateContributionList();
          expect(result.length).toEqual(3)
          expect(result[0].props.children[0].props.children[0]).toEqual('david pac')
      })
    })
  })

  describe('mapStateToProps', () => {
    it('should return a candidate object and a committee_id', () => {
      const mockState = {
        candidate: { name: 'Dorvid' },
        committee_id: 'ABC123'
      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState)
    })
  })
})