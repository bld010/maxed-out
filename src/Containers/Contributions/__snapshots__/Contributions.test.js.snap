// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributions should match the snapshot with correct data passed in 1`] = `
<section
  className="Contributions"
>
  <h3>
    PAC
     Contributions
  </h3>
  <h4>
    No 
    PAC
     contributions found for the 2020 election cycle.
  </h4>
</section>
`;
